app/controllers/user_profiles_controller.rb

```diff
@@ -38,9 +33,11 @@ def show
   end
 
   def edit
-    @user = User.last
-    # @user_profile = current_user.user_profile
-    @user_profile = @user.user_profile
+    if current_user.profile_registered?
+      @user_profile = current_user.user_profile
+    else
+      redirect_to register_path
+    end
```

Comment:
ここも

```suggestion
    if !current_user.profile_registered?
      redirect_to register_path
    end

    @user_profile = current_user.user_profile
```

とかできると、シンプルになりそうですね。

Reply:
Rubcopが通らない恐れがあります。そのままにします。